### PR TITLE
test: Add Travis-CI continuous integration

### PR DESCRIPTION
**Pull request by Echelon9**
Configure, build and run test suite on GCC and Clang.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+language: cpp
+
+# Use the faster container-based infrastructure.
+sudo: false
+
+# Use Trusty beta build environment.
+dist: trusty
```
Comment (karolherbst) [inline .travis.yml]: why trusty? If there is no reason then leave it out

Comment (Echelon9) [inline .travis.yml]: Travis' default containerized image ships with an older Ubuntu, and relevantly a very old GCC (4.6.3) and Clang (3.4). These compilers fail on envytools as they don't fully support C++11.

So the options were either to (a) add to envytools' travis.yml lots of configuration commands to update to more modern compiler releases; or (b) base off Travis' more modern Ubuntu images that have the updated compilers already.

The (b) approach chosen here is far more terse and also faster on each run to achieve the same outcome.

Note: Travis don't yet provide containerized images more modern than trusty.

Comment (karolherbst) [inline .travis.yml]: I see

**Comment by Echelon9** [comment]
@karolherbst any chance you might merge this? Any further comments that you would like to see addressed first?

**Comment by karolherbst** [comment]
there are three things left inside the output:
* Warning: nvapy won't be built because of un-met dependencies (python and cython)
* warning: cupti_trace won't be built because of un-met dependencies (cuda and/or valgrind-mmt)
* Warning: demmt won't sandbox itself because libseccomp was not found

Would be nice to also build test those features, allthough I see that cuda/valgrind-mmt are hard to setup in travis, so we can leave out the second warning for now.

**Comment by Echelon9** [comment]
Hi,

#1: I planned to address this in a follow up PR, as this warning exposed that the documentation and build requirements could be improved on this point (python isn't listed as an nva dependency in all spots at present). Given that PR would then start to touch more than a travis file, I thought better to not confuse too many changes together.

#3: `libseccomp-dev` is not on the Travis apt whitelist for the containerized environments (see https://github.com/travis-ci/apt-package-whitelist/blob/master/ubuntu-trusty and https://github.com/travis-ci/apt-package-whitelist/issues/1909). There are enough benefits of the containerized environment, e.g. speed, to make staying with it worthwhile. Note: we don't get `sudo` access to install it directly. Travis does over time add new packages to whitelists, so the situation may improve in the future.

Do you agree?

**Comment by karolherbst** [comment]
I would add the deps for #1 already, because then you don't need to touch the .travis.yml file anymore.

Regarding #3 we can wait until the packages are whitelisted. I think you could actually request that package, so that they ship it faster.

I wouldn't block this PR though because of #1, but I would really do it in the first version already, because we know about it.

**Comment by karolherbst** [comment]
@Echelon9 any thoughts on that? I will merge either version, but I would really prefer one with the deps already included.

**Comment by Echelon9** [comment]
I'll be able to add the #1 deps to this PR in a few hours.

**Comment by Echelon9** [comment]
Upon further work, it appears `cython` and `cython3` are presently not white listed by Travis-CI for the Ubuntu image we propose using (https://github.com/travis-ci/apt-package-whitelist/issues/382). So that's a dependency halt on #1.

I've added an extra comment to both so Travis-CI is aware of our request for each of the dependencies to be added.

For now, the extra documentation updates and comments sprinkled throughout are as good as we seem to be able to get at the moment. I'd suggest that this PR is now merged as is, and I can follow up later on when the dependencies are white listed.

**Pull request merged**